lib/ControlledVocab/ControlledVocab.js

```diff
@@ -51,22 +64,32 @@ class ControlledVocab extends React.Component {
               return `(${resourceValues.updaterIds.join(' or ')})`;
             }
             return null;
-          },
+          }
         }
       }
     },
     updaterIds: {},
+    // Translated library-defined controlled vocabularies [UXPROD-3148]
+    translations: {
+      type: 'okapi',
+      path: 'translations',
```

Comment:
I disagree with this proposed architecture.

The translations of controlled vocabulary should not be stored in a central translation module.
Instead the translations should live in the same back-end module where the non-translated controlled vocabulary lives.

This allows to have self-contained modules with as few inter-module dependencies as possible. In a micro-services-like architecture modules should be as independent as possible.

This matches the Technical Council's decision that back-end modules should handle i18n/l10n of messages: https://wiki.folio.org/display/DD/Handle+i18n+where+messages+are+generated

Does it have advantages that the translations of controlled vocabulary live in a different module?

Reply:
There are many advantages that push us to save translations in a different module that I will explain in the translation app docs soon, but as a quick answer I will mention some of the risks and challenges that you will face if we are saving translations in the same back-end module:

1- It will require many changes for all back-and-modules that have controlled vocabulary. (Our solution does not require any changes at the level of back-and-modules)

2- Folio supports more than 26 languages ​​so far, which means that we are in the process of complicated multi-lingual schema and dramatic changes on the back-end architecture level are needed.

3- The amount of changes required on the stripes app to display the translated strings ​​instead of the original values ​​will be very large (our solution does not require this size of changes because all translations are fetched from one place and then merged with the default translations which are handled by react -intl )

About the Technical Council's decision that back-end modules should handle i18n/l10n of messages:
I think this is a different case because it is concerned with the error and validation messages generated by developers in the backend and I think they will be static translations (I don't know what is the approach the community follow-up to solve this issue).

Reply:
This is an important discussion but I don't think this PR is the right place for it. 

@attia-alshareef you and your team have done some impressive work here but before we can start merging PRs, we need to make sure that the specific shape of the feature (l10n of runtime values) is acceptable to the Product Council and that its technical implementation passes muster with the Tech Leads (or Tech Council, if the folks at Tech Leads decide they need to boot it up a level for official approval). We are in the process of organizing that now. 

In any case, this UI work should not be merged until the other services that support it -- i.e. new backend modules, or changes to the interfaces of existing modules -- is already in place.